ffi/newpassmanagers.cpp

```diff
@@ -0,0 +1,317 @@
+#include "core.h"
+#include "llvm-c/TargetMachine.h"
+#include "llvm/Analysis/AliasAnalysisEvaluator.h"
+#include "llvm/IR/PassManager.h"
+#include "llvm/IR/Verifier.h"
+#include "llvm/Passes/PassBuilder.h"
+#include "llvm/Transforms/InstCombine/InstCombine.h"
+#include "llvm/Transforms/Scalar/JumpThreading.h"
+#include "llvm/Transforms/Scalar/LoopRotation.h"
+#include "llvm/Transforms/Scalar/LoopUnrollPass.h"
+#include "llvm/Transforms/Scalar/SimplifyCFG.h"
+
+using namespace llvm;
+
+/*
+ * Exposed API
+ */
+
+namespace llvm {
+
+struct OpaqueModulePassManager;
+typedef OpaqueModulePassManager *LLVMModulePassManagerRef;
+DEFINE_SIMPLE_CONVERSION_FUNCTIONS(ModulePassManager, LLVMModulePassManagerRef)
```

Comment:
We should probably be using this macro whenever we want `wrap()` and `unwrap()` functions elsewhere in llvmlite - I am not sure why we're not doing that already.

Reply:
yes I thought the same, will take a look.

Reply:
Ah, sorry, I didn't mean to imply you should sort that out in this PR. It could/should be orthogonally done in another PR rather than piling more changes in here.